source/simulator/parameters.cc

```diff
@@ -764,12 +764,19 @@ namespace aspect
                   << "-----------------------------------------------------------------------------\n\n"
                   << std::endl;
 
-        // create the directory. we could call the 'mkdir()' function directly, but
-        // this can only create a single level of directories. if someone has specified
+        // Create the directory. We could call the 'mkdir()' function directly, but
+        // this can only create a single level of directories. If someone has specified
         // a nested subdirectory as output directory, and if multiple parts of the path
-        // do not exist, this would fail. working around this is easiest by just calling
-        // 'mkdir -p' from the command line
-        const int error = system ((std::string("mkdir -p '") + output_directory + "'").c_str());
+        // do not exist, this would fail. Working around this is easiest by just calling
+        // 'mkdir -p' from the command line.
+        int error = system ((std::string("mkdir -p '") + output_directory + "'").c_str());
+
+        // If no command-line exists (such as on compute nodes of IBM BlueGene/Q), or
+        // mkdir is not available on the command-line, fall back to mkdir().
+        if (error != 0)
+          {
+            error = mkdir(output_directory.c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
```

Comment:
The difficulty that prompted me to use the `system("mkdir -p")` call is that the `mkdir` function can only create one level of subdirectories, if I recall this correctly. On the other hand, `mkdir -p` creates as many levels as necessary.

The distinction is relevant in cases where someone sets the output directory to `./output/run-13/test-42`, but neither `output/` doesn't exist yet. This is slightly complicated with just `std::mkdir` -- a solution is here, for example: http://stackoverflow.com/questions/675039/how-can-i-create-directory-tree-in-c-linux . (It would be nice if we could use the boost::filesystem function, but that's not part of our piece of Boost :-( . There is a nice algorithm further down that page, posted by user "Patrick", though.)